enclave/src/main/java/org/hyperledger/besu/enclave/EnclaveFactory.java

```diff
@@ -43,4 +53,57 @@ public Enclave createVertxEnclave(final URI enclaveUri) {
 
     return new Enclave(vertxTransmitter);
   }
+
+  public Enclave createVertxEnclave(
+      final URI enclaveUri,
+      final Path orionKeyStoreFile,
+      final Path orionKeyStorePasswordFile,
+      final Path orionWhitelistFile) {
+    if (enclaveUri.getPort() == -1) {
+      throw new EnclaveIOException("Illegal URI - no port specified");
+    }
+
+    final HttpClientOptions clientOptions = new HttpClientOptions();
+    clientOptions.setDefaultHost(enclaveUri.getHost());
+    clientOptions.setDefaultPort(enclaveUri.getPort());
+    clientOptions.setConnectTimeout(CONNECT_TIMEOUT);
```

Comment:
This is duplicated with the non-tls create method, is it possible to remove this duplication at all?

Reply:
done